### ISSUE TITLE
Gatsby documentation site sometimes breaks, showing only a blank page

### ISSUE THREAD

**Issue by arggh** [gatsbyjs/gatsby#5032]
## Description

I've been trying out Gatsby and thus reading a lot of the documentation. Quite often the documentation site breaks for me, if I leave it open for a while and then later try to navigate.

### Steps to reproduce

1. I have a Gatsby documentation open in Chrome tab
2. I leave it open for a while
3. I get back to it & search for something
4. Click the search result
=> Empty blank page & JavaScript error

### Expected result

A site should work like a website does, so either open the page I was trying to access or show some kind of error message explaining that things didn't go too well.

### Actual result

![gatsbyfail](https://user-images.githubusercontent.com/17210302/38918870-9ed218ca-42f7-11e8-8260-fb94f3a1a29d.png)

Blank white page and JavaScript error in console (the red box is from a Chrome extension which alerts  about JavaScript errors)

### Environment

* Operating System: Mac OS 10.12.6 & Chrome 65

**Comment by m-allanson** [comment]
Hey @arggh thanks for reporting this, I'd like to try and replicate it so I've got some questions... 

how long is a while? Does this _always_ happen after a set amount of time? Does it only happen for some pages or search terms? Are there any other errors in the console?

**Comment by arggh** [comment]
I'll try to keep an eye out for these incidents and check the console for more details next time.

I just came back from the grocery store and had 4 tabs open at different pages on Gatsbyjs.org, so I'd say I didn't touch any of the tabs in approx. 45 minutes. I then tried to use the search and navigate on each one of them, all worked except for one:

![gatsby_site_error1](https://user-images.githubusercontent.com/17210302/38945437-2af5302a-433f-11e8-8533-9cc320d8f1a4.png)

I don't know if this is a different issue, since there was no blank page this time - the page just remained unresponsive and clicking on links didn't have any visible effect.
